Orange/widgets/data/owfile.py

```diff
@@ -470,6 +470,10 @@ def _get_reader(self) -> FileFormat:
             return reader
         else:
             url = self.url_combo.currentText().strip()
+            # tolerant parse, also prevent parsing as local file
+            parsed = QUrl.fromUserInput(url, os.devnull)
+            if parsed.isValid():
+                url = bytes(parsed.toEncoded()).decode("ascii")
```

Comment:
The problem of this solution is, that the UrlReader itself still doesn't work, so the change would have to be made in every widget that uses it.

I'd rather the fix was made within the UrlReader.

Reply:
No. UrlReader should not accept invalid urls and should not do anything that could break valid urls.

Reply:
Yes, but valid at which level? Shouldn't we support URLs as a browser would? If there is some non-destructive encoding/decoding happening in the background, why not put it into `UrlReader`?

Reply:
Fine.